test/e2e/delete_test.go

```diff
@@ -112,6 +115,16 @@ func testWindowsNodeDeletion(t *testing.T) {
 	testCtx, err := NewTestContext()
 	require.NoError(t, err)
 
+	// Deploy a DaemonSet and wait until its pods have been made ready on each Windows node. DaemonSet pods cannot be
+	// drained from a Node.
+	// Doing this ensures that WMCO is able to handle containers being present on the instance when deconfiguring.
+	ds, err := testCtx.deployNOOPDaemonSet()
+	require.NoError(t, err, "error creating DaemonSet")
+	defer testCtx.client.K8s.AppsV1().DaemonSets(ds.GetNamespace()).Delete(context.TODO(), ds.GetName(),
```

Comment:
Why are you deleting the `DaemonSet`? In the real world case nothing is going to delete the `DaemonSet`. Are you just doing this to do cleanup after all the tests have finished running?

Reply:
Yes this is just cleaning up after test completion.